crates/net/network/src/config.rs

```diff
@@ -473,7 +469,11 @@ impl NetworkConfigBuilder {
     /// The given client is to be used for interacting with the chain, for example fetching the
     /// corresponding block for a given block hash we receive from a peer in the status message when
     /// establishing a connection.
-    pub fn build<C>(self, client: C) -> NetworkConfig<C> {
+    pub fn build<C>(
+        self,
+        client: C,
+        chain_spec: impl EthChainSpec + Hardforks,
```

Comment:
I've tried this initially but it was causing issues for some examples with noop provider because they required a different chainspec vs the one it was returning

I've updated `NoopBlockReader` and `build_with_noop_provider` to accept a chainspec in f89cefd, so the additional chainspec arg is not needed most of the time now